uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/BasemapGalleryController.qml

```diff
@@ -221,6 +248,21 @@ QtObject {
             }
         }
 
+        function setCurrentBasemapMatchingSp(basemap, layer) {
+            if(layer.loadStatus === Enums.LoadStatusLoaded) {
+                if(!basemapMatchesCurrentSpatialReference(basemap)){
+                    //manually setting the currbasemap to trigger the event onCurrentBasemapChaged
+                    internal.currentBasemap = internal.currentBasemap;
+                    geoModel.basemap = geoModel.basemap;
+                    internal.gallery.dataChanged(internal.gallery.index(basemapIndex(basemap), 0), internal.gallery.index(basemapIndex(basemap), 0));
+                    return;
+                }
+                internal.currentBasemap = basemap;
+                geoModel.basemap = internal.currentBasemap;
+                layer.loadStatusChanged.disconnect(setCurrentBasemapMatchingSp);
```

Comment:
🔧 An early return may prevent this disconnet from being called, I suggest moving this to the begining of the function like so:

```qml
if (layer.loadStatus === Enums.LoadStatusLoaded) {
    layer.loadStatusChanged.disconnect(setCurrentBasemapMatchingSp);
    ....
```

I used to work for a company that disallowed return statements as part of control-flow. While this is not a requirement at Esri, I would encourage being sparse with this style so that these kinds of bugs can be avoided.